File: generate_mermaid_chart.py
import os

ignored_directories = ["in-progress"]
root_directories = [d for d in os.listdir('.') if os.path.isdir(d) and d not in ignored_directories and d.lower() != "readme.md"]

mermaid_code = "graph LR\n"
for directory in root_directories:
    mermaid_code += f"    {directory} --> {directory}_content[\"{directory}\"]\n"

print(mermaid_code)

File: generate_mermaid_chart.js
const fs = require('fs');
const path = require('path');

const ignoredDirectories = new Set(['in-progress', 'node_modules', '.git']);
const rootPath = '.'; // Root of the repository

// Function to scan directories and filter based on criteria
function scanDirectories(basePath) {
    return fs.readdirSync(basePath, { withFileTypes: true })
        .filter(dirent => dirent.isDirectory() && !ignoredDirectories.has(dirent.name))
        .map(dirent => dirent.name);
}

// Generate Mermaid chart code
function generateMermaidChart(directories) {
    let mermaidCode = 'graph LR\n';
    directories.forEach(dir => {
        mermaidCode += `    ${dir} --> ${dir}_content["${dir}"]\n`;
    });
    return mermaidCode;
}

const directories = scanDirectories(rootPath);
const mermaidChart = generateMermaidChart(directories);
console.log(mermaidChart);
